services/network_config_manager/src/main/java/com/futurewei/alcor/netwconfigmanager/client/gRPC/GoalStateClientImpl.java

```diff
@@ -61,6 +75,7 @@ public GoalStateClientImpl() {
                 new DefaultThreadFactory("grpc-thread-pool"));
 
         //TODO: Setup a connection pool. one ACA, one client.
+        this.hostIpGrpcChannelStubMap = new TreeMap<>();
```

Comment:
TreeMap is an implementation that is not synchronized (Ref: https://docs.oracle.com/javase/8/docs/api/java/util/TreeMap.html)

Do you consider ConcurrentHashMap?

Reply:
> TreeMap is an implementation that is not synchronized (Ref: https://docs.oracle.com/javase/8/docs/api/java/util/TreeMap.html)
> 
> Do you consider ConcurrentHashMap?

Changed this map to ConcurrentHashMap.